pkg/engine/engine.go

```diff
@@ -399,3 +411,17 @@ func isTemplateValid(ch *chart.Chart, templateName string) bool {
 func isLibraryChart(c *chart.Chart) bool {
 	return strings.EqualFold(c.Metadata.Type, "library")
 }
+
+func readTemplateMagicComments(templateBody string) templateOpts {
+	templateOpts := templateOpts{}
+	matches := magicCommentsRegexp.FindAllSubmatch([]byte(templateBody), -1)
+	for _, match := range matches {
+		if strings.EqualFold(string(match[1]), "delim") {
+			delim := strings.SplitN(string(match[2]), ",", 2)
+			templateOpts.delimL = strings.Repeat(delim[0], 2)
+			templateOpts.delimR = strings.Repeat(delim[1], 2)
```

Comment:
Should there be a length check in place? What happens if I write the following?

```
# helm: delim=[
```

Wouldn't that result in a `panic()`?

Reply:
Yes, good catch!
This makes me wonder if it may be better to use separate modifiers for each delimiter like the text/template` does:

```
# helm: delimL=[
# helm: delimR=]
```

I'll give it another thought and play with the code to see the pros/cons

Reply:
fixed in new code